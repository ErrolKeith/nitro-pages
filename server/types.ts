export interface BusinessClientGlobals {
  businessId: string; //uuidv4
  sites: string[]; //uuidv4[]
  companyName: string;
  phoneNumber: string;
}

export interface SiteGlobals {
  businessId: string; //uuidv4
  siteId: string;
  companyName: string;
  phoneNumber: string;
  blogSettings: {
    blogTitle: string;
  };
}
