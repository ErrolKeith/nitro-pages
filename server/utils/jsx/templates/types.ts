const templates = ["home", "page", "internal", "404", "index", "post"] as const;
type TemplateType = (typeof templates)[number];

export interface TemplateConfig {
  type: TemplateType;
  siteId: string;
  companyName: string;
  seoTitle: string;
  content: PageContentSection[];
}
