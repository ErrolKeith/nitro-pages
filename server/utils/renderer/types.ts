import { SiteGlobals } from "~/types";
import { TemplateType } from "../jsx/template";

export interface PageContext {
  template: TemplateType;
  slug: string;
  title: string;
  categories: string[];
  content: PageContentSection[];
}

export interface BaseRenderConfig {
  template: TemplateType;
  clientGlobals: SiteGlobals;
  pageContext: PageContext;
}

export interface PageRenderConfig extends BaseRenderConfig {
  template: "page" | "internal" | "post" | "index";
}

export type HomeRenderConfig = Omit<BaseRenderConfig, "template">;
