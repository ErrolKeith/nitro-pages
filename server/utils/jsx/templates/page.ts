import { TemplateConfig } from "./types";
import { makeHtml } from "../tags/html";
import { makeHead } from "../tags/head";
import { makeMain } from "../tags/main";
import { makeTitle } from "../tags/title";

export function makePageTemplate(config: TemplateConfig) {
  const { siteId, seoTitle, content } = config;

  return makeHtml({
    props: {
      ["data-site-id"]: siteId,
    },
    children: [
      makeHead({ props: {}, children: [makeTitle({ title: seoTitle })] }),
      makeMain({ props: {}, children: [] }),
    ],
  });
}
