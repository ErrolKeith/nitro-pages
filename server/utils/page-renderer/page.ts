import { RenderConfig } from "../../types";
import Handlebars from "handlebars";

const pageTemplate = Handlebars.compile(`
  <head>
    <title>{{ title }}</title>
  </head>
  <body data-site-id='{{ siteId }}'>
    <h1>{{ heading }}</h1>
    <main>
    <div class="hero">{{ heroText }}</div>
    <div>
      <p>
        Content
      </p>
    </div>
    </main>
  </body>
`);

const pageNotFoundTemplate = Handlebars.compile(`
  <head>
    <title>Page Not Found | {{ companyName }}</title>
  </head>
  <body data-site-id='{{ siteId }}'>
    <h1>404</h1>
    <main>
    <div class="hero">Page Not Found</div>
    <div>
      <p>
        Content
      </p>
    </div>
    </main>
  </body>
`);

function render(config: RenderConfig) {
  if (config.type === "page") {
    return pageTemplate({
      siteId: config.clientGlobals.siteId,
      title: `${config.clientGlobals.companyName} | ${config.pageContext.title}`,
      heading: config.clientGlobals.companyName,
      heroText: config.pageContext.heroText,
    });
  }

  if (config.type === "service") {
    return pageTemplate({
      siteId: config.clientGlobals.siteId,
      title: `${config.clientGlobals.companyName} | ${config.pageContext.title}`,
      heading: config.pageContext.title,
      heroText: config.pageContext.heroText,
    });
  }

  return pageNotFoundTemplate({
    siteId: config.clientGlobals.siteId,
    companyName: config.clientGlobals.companyName,
  });
}

export { render };
