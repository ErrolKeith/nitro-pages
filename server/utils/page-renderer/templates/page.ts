import Handlebars from "handlebars";

export const pageTemplate = Handlebars.compile(`
  <head>
    <title>{{ seoTitle }}</title>
  </head>
  <body data-site-id='{{ siteId }}'>
    <main>
        <h1>{{ title }}</h1>
    </main>
  </body>
`);
