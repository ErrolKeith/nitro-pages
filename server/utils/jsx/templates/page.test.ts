import { it, expect } from "vitest";
import { makePageTemplate } from "./page";
import { TemplateConfig } from "./types";

it("can make a page", () => {
  const testPageTemplateConfig: TemplateConfig = {
    type: "internal",
    siteId: "124246796763",
    companyName: "sample company name",
    seoTitle: "sample internal",
    content: [],
  };
  const templateInternalTest = makePageTemplate(testPageTemplateConfig);
  expect(templateInternalTest).toMatchSnapshot();
});
