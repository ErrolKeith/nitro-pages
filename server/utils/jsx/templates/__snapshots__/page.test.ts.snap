// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make a page 1`] = `
<html
  data-site-id="124246796763"
>
  <head>
    <title>
      sample internal
    </title>
  </head>
  <main />
</html>
`;
